frontend/src/old-pages/Clusters/Instances.tsx

```diff
@@ -42,77 +42,94 @@ import {InstanceStatusIndicator} from '../../components/Status'
 import EmptyState from '../../components/EmptyState'
 import DateView from '../../components/date/DateView'
 
-function InstanceActions({
-  fleetStatus,
-  instance,
-}: {
-  fleetStatus: ComputeFleetStatus
-  instance: Instance
-}) {
-  const pending = useState(['app', 'clusters', 'action', 'pending'])
-  const clusterName = useState(['app', 'clusters', 'selected'])
-  const navigate = useNavigate()
-  const logHref = `/clusters/${clusterName}/logs?instance=${instance.instanceId}`
+function InstanceActions({instance}: {instance?: Instance}) {
   const {t} = useTranslation()
+  const navigate = useNavigate()
+
+  const clusterName = useState(['app', 'clusters', 'selected'])
+  const fleetStatus: ComputeFleetStatus = useState([
+    'clusters',
+    'index',
+    clusterName,
+    'computeFleetStatus',
+  ])
+  const logHref = `/clusters/${clusterName}/logs?instance=${
+    instance && instance.instanceId
+  }`
+  const [startPending, setStartPending] = React.useState(false)
+  const [stopPending, setStopPending] = React.useState(false)
+
+  const resetPending = () => {
+    setStopPending(false)
+    setStartPending(false)
+  }
 
   const refresh = () => {
     const clusterName = getState(['app', 'clusters', 'selected'])
     clusterName &&
-      GetClusterInstances(clusterName, () =>
-        clearState(['app', 'clusters', 'action', 'pending']),
-      )
+      GetClusterInstances(clusterName, () => {
+        resetPending
+      })
```

Comment:
issue: since `resetPending` is never called, the button will always be stuck in a loading state